src/App.tsx

```diff
@@ -1,61 +1,175 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { User } from './type/User';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+function checkKey() {
+  const maxId = [...todos].sort((a, b) => b.id - a.id);
+
+  for (let i = 1; i < maxId[0].id; i += 1) {
+    if (todos.every(todo => todo.id !== i)) {
+      return i;
+    }
+  }
+
+  console.log(todos); // eslint-disable-line
+
+  return maxId[0].id + 1;
+}
```

Comment:
I'm not sure, but I think you don't need to check the last id every time. You can check it only once and save it into a variable, and when you create a new todo just increase this id number. I did that😅

Reply:
> I'm not sure, but I think you don't need to check the last id every time. You can check it only once and save it into a variable, and when you create a new todo just increase this id number. I did that😅

not last id, it is the id that is not in the array